lib/core/include/data_object_proxy.hpp

```diff
@@ -0,0 +1,212 @@
+        {
+            // TODO: should this set the logical path too?
+            for (auto& r : r_) {
+                r.collection_id(_cid);
+            }
+        }
+
+        /// \brief Set the logical path for the data object (i.e. all replicas)
+        /// \param[in] _lp - New value for logical path
+        /// \returns void
+        /// \since 4.2.9
+        template<
+            typename P = doi_type,
+            typename = std::enable_if_t<!std::is_const_v<P>>>
+        auto logical_path(std::string_view _lp) -> void
+        {
+            // TODO: should this set the collId too?
+            for (auto& r : r_) {
+                r.logical_path(_lp);
+            }
+        }
+
+        /// \brief Set the owner user name for the data object (i.e. all replicas)
+        /// \param[in] _un - New value for owner user name
+        /// \returns void
+        /// \since 4.2.9
+        template<
+            typename P = doi_type,
+            typename = std::enable_if_t<!std::is_const_v<P>>>
+        auto owner_user_name(std::string_view _un) -> void
+        {
+            for (auto& r : r_) {
+                r.owner_user_name(_un);
+            }
+        }
+
+        /// \brief Set the owner zone name for the data object (i.e. all replicas)
+        /// \param[in] _zn - New value for owner zone name
+        /// \returns void
+        /// \since 4.2.9
+        template<
+            typename P = doi_type,
+            typename = std::enable_if_t<!std::is_const_v<P>>>
+        auto owner_zone_name(std::string_view _zn) -> void
+        {
+            for (auto& r : r_) {
+                r.owner_zone_name(_zn);
+            }
+        }
+
+        /// \brief Get reference to replica list
+        /// \returns replica_list&
+        /// \since 4.2.9
+        template<
+            typename P = doi_type,
+            typename = std::enable_if_t<!std::is_const_v<P>>>
+        auto replicas() -> replica_list& { return r_; }
+
+        /// \brief Get pointer to the underlying dataObjInfo_t struct
+        /// \returns doi_pointer_type
+        /// \since 4.2.9
+        template<
+            typename P = doi_type,
+            typename = std::enable_if_t<!std::is_const_v<P>>>
+        auto get() -> doi_pointer_type { return doi_; }
+
+    private:
+        /// \brief Pointer to underlying doi_type
+        /// \since 4.2.9
+        doi_pointer_type doi_;
+
+        /// \brief List of objects representing physical replicas
+        /// \since 4.2.9
+        replica_list r_;
+
+        /// \brief Generates replica proxy objects from doi_type's linked list and stores them in a list
+        /// \since 4.2.9
+        static auto fill_replica_list(doi_type& _doi) -> replica_list
+        {
+            replica_list r;
+            for (doi_type* d = &_doi; d; d = d->next) {
+                r.push_back(replica::replica_proxy{*d});
+            }
+            // TODO: do we allow empty replica list?
```

Comment:
no?    what's the use case for that?    can anyone hold it wrong?

Reply:
Hmmm... you're right. I guess it doesn't make any sense - it has to be at least 1 replica long as the very existence of a `dataObjInfo_t` suggests a replica. I will remove this